statsmodels/discrete/discrete_model.py

```diff
@@ -2829,12 +2812,8 @@ def fit_regularized(self, start_params=None, method='l1',
                 full_output=full_output, disp=disp, callback=callback,
                 alpha=alpha, trim_mode=trim_mode, auto_trim_tol=auto_trim_tol,
                 size_trim_tol=size_trim_tol, qc_tol=qc_tol, **kwargs)
-        if method in ['l1', 'l1_cvxopt_cp']:
-            discretefit = L1NegativeBinomialResults(self, cntfit)
-        else:
-            raise Exception(
```

Comment:
Duplicates check in DiscreteModel.fit_regularized.  `Exception` isn't as bad as `TypeError` below, but still not great.